### PR TITLE
Backend: PlayerDeathEvent

### PR DESCRIPTION
**Pull request by hannibal002**
## What
Creating and using PlayerDeathEvent.

## Changelog Technical Details
+ Created and implemented PlayerDeathEvent. - hannibal2

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        deathMessagePattern.matchMatcher(message) {
+            val name = group("name")
+            val reason = group("reason").removeColor()
+            PlayerDeathEvent(name, reason, event).postAndCatch()
+
+        }
+    }
+}
+
+class PlayerDeathEvent(val name: String, val reason: String, val chatEvent: LorenzChatEvent) : LorenzEvent()
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt]: This should be moved to its own class like all other event

File: src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message
+        deathMessagePattern.matchMatcher(message) {
+            val name = group("name")
+            val reason = group("reason").removeColor()
+            PlayerDeathEvent(name, reason, event).postAndCatch()
+
+        }
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt]:
```suggestion
            PlayerDeathEvent(name, reason, event).postAndCatch()
        }
```

File: src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt [hunk 3]
```diff
@@ -57,11 +52,13 @@ object MiningAPI {
...
         }
-        coldResetDeath.matchMatcher(event.message) {
-            if (group("name") == LorenzUtils.getPlayerName()) {
-                updateCold(0)
-                lastColdReset = SimpleTimeMark.now()
-            }
+    }
+
+    @SubscribeEvent
+    fun onPlayerDeath(event: PlayerDeathEvent) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt:135]: we could early return if the player name is not the user but not necessary

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

File: src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt (hunk 3 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt:135]: early return is not needed and rather confusing when its the only check in the function imo

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/data/MiningAPI.kt:135]: also this is no precondition, but THE condition

File: src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt]: i dont like the current design of all events in one event package. this only provides us with one long list, without order.
rather have the events where they are getting fired or use a more strict sub package strcuture, what do you thing?

Comment (CalMWolfs) [reply src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt]: I think for the sake of consistency this should be moved. I like the idea of specific sub packages and I do plan on doing something about that in the next few weeks but for now we should stick to the existing system

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/data/PlayerDeathManager.kt]: agreed

**Pull request merged**